src/background/listener/bee-api.listener.ts

```diff
@@ -276,8 +276,25 @@ export class BeeApiListener {
    * @param bzzReference in form of $ROOT_HASH<$PATH><$QUERY>
    * @param tabId the tab will be navigated to the dApp page
    */
-  private redirectToBzzReference(bzzReference: string, tabId: number) {
-    const url = `${this._beeApiUrl}/bzz/${bzzReference}`
+  private redirectToBzzReference(bzzReference: string, tabId: number, preventSubdomainRedirection = false) {
+    let url: string
+
+    if (preventSubdomainRedirection || !isLocalhost(this._beeApiUrl)) {
+      url = `${this._beeApiUrl}/bzz/${bzzReference}`
+    } else {
+      const [hash, path] = bzzReference.split(/\/(.*)/s)
+      let subdomain = hash
+
+      if (subdomain.endsWith('.eth')) {
+        subdomain = subdomain.substring(0, subdomain.length - 4)
+      }
+
+      url = createSubdomainUrl(this._beeApiUrl, subdomain)
```

Comment:
here the subdomain is a swarm hash, but u don't transform it into a CID that the swarm.localhost subdomain requires.